Automattic/pocket-casts-android
# Enable Automotive release builds in Fastlane

The main goal of this PR is to enable `automotive` builds in Fastlane which were previously disabled to simplify the implementation & testing. However, it also includes some minor changes/fixes as we are wrapping up the new release process.

* Moves app signing configuration to `base.gradle` so we don't have to duplicate the implementation in `app/build.gradle` & `automotive/build.gradle`
* Moves `secretProperties` & `versionProperties` logic to `dependencies.gradle` and simplifies the implementation. This change made sense to me in the current project structure as I felt that implementing it in `build.gradle` was a small mistake.
* Implements `versionCode` calculation where `automotiveVersionCode = versionCode + 50000`. As far as I can tell from Play Store, this is historically how the `versionCode`s have been separated between `app` & `automotive` and only recently slightly changed when I accidentally pushed a `versionCode` to `Internal Track`. I've mixed feelings about this, and I spent a lot of time looking into different solutions. At the end I regretted the time spent as keeping the current logic is the most practical path at the moment. If we are going to change it, it makes more sense to not combine it with a whole set of other changes.
* Adds `IS_AUTOMOTIVE_BUILD` Gradle property which adds to `50000` to the `versionCode`. Current logic uses environment variables to make a similar change and I don't think either of these approaches are correct. Ideally the `versionCode` is separated in `app/build.gradle` & `automotive/build.gradle` files and wouldn't be impacted by the environment or how the app is run. However, the `versionCode` is added as a `BuildConfig` value to the internal modules that are part of both `app` & `automotive` builds. I spent some time looking into this in f576e2358ef811eecda3470e0f631dad86457293 & b6cbad1a1b21ed1786dd3056778b5f0e3ecba644 (not included in this PR) which limits the `BuildConfig` requirement to the `Settings`, however this is also a part of internal module and passing the version information to it would require a more significant refactor than I thought would be reasonable as part of the release process work. So, I settled on this approach and left a detailed comment about this for future work.
* Replaces `GHHELPER_REPO` which no longer exists with `GH_REPOSITORY` and `create_release` with `create_gh_release`. Both of these were oversights in previous PRs.
* Adds the `IS_AUTOMOTIVE_BUILD` Gradle property to `build_bundle` so that this information will be available for all `fastlane` builds.

### To test

Similar to previous PRs, I don't think it makes much sense to test these changes in isolation and we should wait to run a full release cycle. However, I've run some tests to confirm the following changes:

* I verified that we are able to upload `automotive` bundle together with the `app` bundle to Play Store.
* The release draft for `automotive` wasn't created, even though there is no fastlane error, but I think this is happening because the `beta` track is currently paused.
* I verified that we can create a GitHub release with both `app` & `automotive` assets.
* I verified in Play Store that the `versionCode` for `automotive` was `appVersionCode + 50000`.

# Checklist
